mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java

```diff
@@ -875,18 +887,22 @@ private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIExc
         }
     }
 
-    private static MCRDerivate retrieveMCRDerivate(MCRObjectID parentObjId, String derIDString)
+    //TODO move into generic package for v1 and v2 REST-API
+    public static MCRObjectID retrieveMCRDerivateID(MCRObjectID parentObjId, String derIDString)
         throws MCRRestAPIException {
+        if(parentObjId == null || derIDString == null) {
+            return null;
+        }
 
         String derKey = "mcr"; // the default value for the key
         if (derIDString.contains(":")) {
             int pos = derIDString.indexOf(":");
             derKey = derIDString.substring(0, pos);
             derIDString = derIDString.substring(pos + 1);
-            if (!derKey.equals("mcr") && !derKey.equals("label")) {
+            if (!derKey.equals("mcr") && !derKey.equals("label") && !derKey.equals("classification")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)